sdk/core/System.ClientModel/src/Pipeline/RetryRequestPolicy.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.ClientModel.Primitives;
+
+public class RetryRequestPolicy : PipelinePolicy
+{
+    private const int DefaultMaxRetries = 3;
+
+    private readonly int _maxRetries;
+
+    public RetryRequestPolicy() : this(DefaultMaxRetries)
+    {
+    }
+
+    public RetryRequestPolicy(int maxRetries)
+    {
+        _maxRetries = maxRetries;
+    }
+
+    public override void Process(PipelineMessage message, PipelineProcessor pipeline)
+
+#pragma warning disable AZC0102 // Do not use GetAwaiter().GetResult().
+        => ProcessSyncOrAsync(message, pipeline, async: false).AsTask().GetAwaiter().GetResult();
+#pragma warning restore AZC0102 // Do not use GetAwaiter().GetResult().
+
+    public override async ValueTask ProcessAsync(PipelineMessage message, PipelineProcessor pipeline)
+        => await ProcessSyncOrAsync(message, pipeline, async: true).ConfigureAwait(false);
+
+    private async ValueTask ProcessSyncOrAsync(PipelineMessage message, PipelineProcessor pipeline, bool async)
+    {
+        if (async)
+        {
+            await pipeline.ProcessNextAsync().ConfigureAwait(false);
+        }
+        else
+        {
+            pipeline.ProcessNext();
+        }
+
+        // If "Should Retry"
+        //    GetNextDelay
+        //    Wait(Delay, CancellationToken)
+        //    If (message.HasRespose)
+        //        Dispose the content stream if applicable
+        //    Increment Retry Count
```

Comment:
Do we want to adopt a convention of putting things in the Message property bag so other policies could peek these values?